src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +121,33 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            return SkipLastIterator(source, Math.Max(count, 0));
+        }
+
+        private static IEnumerable<TSource> SkipLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            var queue = new Queue<TSource>();
```

Comment:
I think the best thing when count is zero or less would be to pass to `Skip()` anyway; at best it passes back the source directly if the source came from Linq (this was deemed a relatively safe case for doing so in such cases, whereas with collections there could be complications due to two enumerables expected to be distinct being the safe) and otherwise would still have a lighter implementation of the same response.